play-media-native/screens/CreateMedia.tsx

```diff
@@ -0,0 +1,212 @@
+              fileType: getFileType(route.params.image),
+              fileUrl: route.params.image?.fileUrl || route.params.image?.uri,
+            }
+          : null
+      );
+    }, [route.params.image])
+  );
+
+  useFocusEffect(
+    useCallback(() => {
+      const unsubscribe = navigation.addListener('beforeRemove', (event) => {
+        // Prevent default behavior of leaving the screen
+        //
+        event.preventDefault();
+
+        navigation.push('DiscardChanges', {
+          message: 'Are you sure you want to discard the new media item or continue editing?',
+          redirectRoute: 'MainTabs',
+          subtitle: 'Discard new media?',
+        });
+      });
+
+      // Make sure to remove the listener
+      // Otherwise, it BLOCKS GOING BACK to MainTabs from a nested screen discard action
+      //
+      return () => {
+        unsubscribe();
+      };
+    }, [navigation])
+  );
+
+  // Hide bottom action buttons if a loading indicator or a toaster is shown
+  useEffect(() => {
+    if (isValidating || showSuccessToast || showErrorToast) {
+      setShouldShowBottomActions(false);
+    } else {
+      setShouldShowBottomActions(true);
+    }
+  }, [isValidating, showSuccessToast, showErrorToast]);
+
+  useEffect(() => {
+    navigation.setParams({
+      title: createdImage?.name || 'Untitled media',
+    });
+  }, [createdImage, navigation]);
+
+  const bottomActions = useMemo(
+    () => (
+      <BottomActions>
+        <Button
+          mode="outlined"
+          labelStyle={styles.buttonLabel}
+          style={styles.button}
+          onPress={handleSaveDraft}
+        >
+          Save Draft
+        </Button>
+        <Button
+          mode="contained"
+          labelStyle={styles.buttonLabel}
+          style={styles.button}
+          onPress={handlePublish}
+        >
+          Publish
+        </Button>
+      </BottomActions>
+    ),
+    [handlePublish, handleSaveDraft]
+  );
+
+  if (!createdImage) {
+    return <Text>Something went wrong!</Text>;
+  }
+
+  return (
+    <KeyboardAwareScreen>
+      <Image
+        source={{ uri: createdImage.fileUrl }}
+        style={[
+          styles.responsiveImage,
+          { aspectRatio: createdImage.fileWidth / createdImage.fileHeight },
+        ]}
+      />
+      <View style={[styles.screenPadding, { paddingVertical: theme.spacing.sm }]}>
+        <InputText
+          containerStyle={styles.inputContainer}
+          title="Title"
+          multiline
+          onChange={onNameChange}
+          value={createdImage?.name || ''}
+        />
+        <InputText
+          containerStyle={styles.inputContainer}
+          title="Description"
+          multiline
+          onChange={onDescriptionChange}
+          value={createdImage?.description || ''}
+        />
+      </View>
+      {isValidating && (
+        <View>
+          <ActivityIndicator size="small" animating />
+        </View>
+      )}
+      <Toast
```

Comment:
Latest design has no toasts, but we can just use the activity indicator and a success message right after in their place.